registration/api/v2/views.py

```diff
@@ -238,6 +240,7 @@ def partial_update(self, request, *args, **kwargs):
         responses={**VALIDATION_RESPONSE, 200: ThemeSchemaSerializer, },
     )
     @action(detail=True, methods=["patch"])
+    @transaction.atomic
```

Comment:
@rusrushal13 
I've just tested this and this change alone won't do proper row-locking of the object retrieved by `self.get_object`.
To achieve that, you'll need to use the `.select_for_update` queryset method ([django docs](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#select-for-update)).
```
application = BetaTestApplication.objects.select_for_update().get(id=pk)
with transaction.atomic:
    # Do atomic operation here
```

Reply:
Added a fix for this using select_for_update and transaction atomic both